icepyx/core/APIformatting.py

```diff
@@ -73,7 +73,7 @@ def _fmt_readable_granules(dset, **kwds):
     # list of readable granule names
     readable_granule_list = []
     # if querying either by 91-day orbital cycle or RGT
-    if "cycles" in kwargs.keys() or "tracks" in kwargs.keys():
+    if "cycles" in kwargs or "tracks" in kwargs:
```

Comment:
I can see the argument for using this simplification (thanks for including the notes about it with the PR). However, I am concerned that this has potential to be confusing to folks newer to Python and ultimately make the code less clear/readable to them. It assumes an awareness that Python will automagically include `.keys()` if this is a dict, which is news to me as of this PR. What's the incentive for using this simplification over explicitly including `.keys()`?

Reply:
I think readability is paramount, and the performance concern is negligible, but want to mention it anyway. Calling `.keys()` is causing some extra overhead to create a [view](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects) of the dictionary's keys before checking.

While I do agree that calling `.keys()` here is more explicit, and [explicit is better than implicit](https://peps.python.org/pep-0020/), "readability counts". Sometimes the most readable thing is the way everyone else is doing it, and I think this is one of those cases. The official Python docs endorse this approach:

>To check whether a single key is in the dictionary, use the [in](https://docs.python.org/3/reference/expressions.html#in) keyword.

https://docs.python.org/3/tutorial/datastructures.html

Reply:
Membership test operations (`item in container`) works for dicts, lists, sets, tuples, and all other classes that implement `.__contains__()`
* https://docs.python.org/3/reference/expressions.html#in
* [`https://docs.python.org/3/reference/datamodel.html#object.__contains__`](https://docs.python.org/3/reference/datamodel.html#object.__contains__)

Reply:
IMO this is a great use case for ChatGPT ("comprehension", in the "list-comprehension" sense, not the cognition sense, of documentation or community coding convention) so I threw this question at it, and it did kind of lie by misquoting the documentation, but it came pretty close to what I consider a good answer.

https://chatgpt.com/share/de22c27b-3370-4051-9d58-b5eeb37b9e09

Reply:
Thanks @cclauss and @mfisher87 for helping me understand this convention!

Reply:
Happy to!